src/applications/caregivers/tests/components/FormFields/AddressWithAutofill.unit.spec.js

```diff
@@ -0,0 +1,146 @@
+  street2: 'Suite 550',
+  city: 'Washington',
+  state: 'DC',
+  postalCode: '20005',
+  'view:autofill': false,
+};
+
+describe('CG <AddressWithAutofill>', () => {
+  it('should render AddressWithAutofill component as a form with checkbox and inputs', () => {
+    const formContext = { reviewMode: false, submitted: undefined };
+    const view = render(
+      <Provider store={mockStore}>
+        <AddressWithAutofill
+          canAutofillAddress={canAutofillAddress}
+          errorSchema={errorSchema}
+          idSchema={idSchema}
+          onChange={onChange}
+          schema={schema}
+          formContext={formContext}
+          formData={formData}
+        />
+      </Provider>,
+    );
+
+    expect(
+      view.container.querySelector('va-checkbox#root_primaryAddress_autofill'),
+    ).to.exist;
+    expect(
+      view.container.querySelector('va-text-input#root_primaryAddress_street'),
+    ).to.exist;
+    expect(
+      view.container.querySelector('va-text-input#root_primaryAddress_street2'),
+    ).to.exist;
+    expect(
+      view.container.querySelector('va-text-input#root_primaryAddress_city'),
+    ).to.exist;
+    expect(view.container.querySelector('va-select#root_primaryAddress_state'))
+      .to.exist;
+    expect(
+      view.container.querySelector(
+        'va-text-input#root_primaryAddress_postalCode',
+      ),
+    ).to.exist;
+  });
+
+  it('should render AddressWithAutofill component as a review field', () => {
+    const formContext = { reviewMode: true, submitted: undefined };
+    const view = render(
+      <Provider store={mockStore}>
+        <AddressWithAutofill
+          canAutofillAddress={canAutofillAddress}
+          errorSchema={errorSchema}
+          idSchema={idSchema}
+          onChange={onChange}
+          schema={schema}
+          formContext={formContext}
+          formData={formData}
+        />
+      </Provider>,
+    );
+
+    // street
+    expect(view.container.querySelectorAll('dt')[0]).to.contain.text(
+      'current street address',
+    );
+    expect(view.container.querySelectorAll('dd')[0]).to.contain.text(
+      '1350 I St. NW',
+    );
+
+    // street2
+    expect(view.container.querySelectorAll('dt')[1]).to.contain.text(
+      'Street address line 2',
+    );
+    expect(view.container.querySelectorAll('dd')[1]).to.contain.text(
+      'Suite 550',
+    );
+
+    // city
+    expect(view.container.querySelectorAll('dt')[2]).to.contain.text('City');
+    expect(view.container.querySelectorAll('dd')[2]).to.contain.text(
+      'Washington',
+    );
+
+    // state
+    expect(view.container.querySelectorAll('dt')[3]).to.contain.text('State');
+    expect(view.container.querySelectorAll('dd')[3]).to.contain.text(
+      'District Of Columbia',
+    );
+
+    // zip
+    expect(view.container.querySelectorAll('dt')[4]).to.contain.text(
+      'Postal code',
+    );
+    expect(view.container.querySelectorAll('dd')[4]).to.contain.text('20005');
+  });
```

Comment:
Can we add `data-testid` attributes to these and query that? I'm not a huge fan of just grabbing from the array by index.